### PR TITLE
PERL-1127 - Support Pipelines in BulkWriteView

### PR DESCRIPTION
**Pull request by TBSliver**
Unless I missed something it was basically already supported - test included!

### CODE REVIEW

File: lib/MongoDB/BulkWriteView.pm [hunk 1]
```diff
@@ -107,7 +107,7 @@ sub _update {
     my ( $self, $doc ) = @_;
 
     my $type = ref $doc;
-    unless ( @_ == 2 && grep { $type eq $_ } qw/HASH ARRAY Tie::IxHash/ ) {
+    unless ( @_ == 2 && grep { $type eq $_ } qw/HASH ARRAY Tie::IxHash BSON::Array/ ) {
         MongoDB::UsageError->throw("argument to $method must be a single hashref, arrayref or Tie::IxHash");
```
Comment (xdg) [inline lib/MongoDB/BulkWriteView.pm]: Please update the error message also.

Comment (TBSliver) [reply lib/MongoDB/BulkWriteView.pm]: Ah! yep will do

**Comment by xdg** [comment]
I'm running one last Evergreen check, then I'll merge this in myself afterwards.

**Comment by xdg** [comment]
Cherry picked as c40e1e14dd648eacbb2a6970b8dbe00cf06b4350

**Pull request closed without merging**